### PR TITLE
Fixed bug which prevent override ProductInventoryChange

### PR DESCRIPTION
**Pull request by asvishnyakov**
Without that serialize adjust inventory background job fails with "unable to cast" exception if you trying override ProductInventoryChange

### CODE REVIEW

File: VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs [hunk 1]
```diff
@@ -151,12 +150,12 @@ public virtual ProductInventoryChange[] GetProductInventoryChangesFor(GenericCha
...
-                    {
-                        ProductId = changedItem.ProductId,
-                        QuantityDelta = newQuantity - oldQuantity,
-                        FulfillmentCenterId = GetFullfilmentCenterForLineItem(changedItem, customerOrder.StoreId, customerOrderShipments)
-                    });
+                    var itemChange = AbstractTypeFactory<ProductInventoryChange>.TryCreateInstance();
+                    itemChange.ProductId = changedItem.ProductId;
+                    itemChange.QuantityDelta = newQuantity - oldQuantity;
+                    itemChange.FulfillmentCenterId = GetFullfilmentCenterForLineItem(changedItem, customerOrder.StoreId,
+                        customerOrderShipments);
```
[the remark this answers is not in the document]
Comment (yecli) [reply VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: Fixed in https://github.com/VirtoCommerce/vc-module-order/commit/b0b730f65a6d8d9a38e8a5ae356174a4b16a74c7

File: VirtoCommerce.OrderModule.Web/JsonConverters/PolymorphicOperationJsonConverter.cs [hunk 2]
```diff
@@ -8,14 +8,20 @@
...
+using VirtoCommerce.OrderModule.Data.Handlers;
 using VirtoCommerce.OrderModule.Data.Utilities;
 using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.OrderModule.Web.JsonConverters
 {
     public class PolymorphicOperationJsonConverter : JsonConverter
     {
-        private static readonly Type[] _knowTypes = { typeof(IOperation), typeof(LineItem), typeof(CustomerOrderSearchCriteria), typeof(PaymentMethod), typeof(ShippingMethod) };
+        private static readonly Type[] _knowTypes =
```
[the remark this answers is not in the document]
Comment (yecli) [reply VirtoCommerce.OrderModule.Web/JsonConverters/PolymorphicOperationJsonConverter.cs]: Fixed in https://github.com/VirtoCommerce/vc-module-order/commit/b0b730f65a6d8d9a38e8a5ae356174a4b16a74c7

File: VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (asvishnyakov) [reply VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: @artem-dudarev No more than 120 symbols :)

[the remark this answers is not in the document]
Comment (artem-dudarev) [reply VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: We don't have such a rule.
And we don't write code on smartphones to introduce this rule :)

**Comment by yecli** [comment]
Released in https://github.com/VirtoCommerce/vc-module-order/releases/tag/v2.17.25

File: VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (asvishnyakov) [reply VirtoCommerce.OrderModule.Data/Handlers/AdjustInventoryOrderChangedEventHandler.cs]: @artem-dudarev We need to discuss about that once upon a time

**Pull request merged**